### ISSUE TITLE
Valgrind warning on arrayobject.c (migrated from Trac #355)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#962]
Original ticket http://projects.scipy.org/numpy/ticket/355
Reported 2006-10-18 by atmention:FrancescAlted, assigned to unknown.

Hi,

I'm trying to run the pytables tests against valgrind and so far, I've got the next warning:

```
==6038== Conditional jump or move depends on uninitialised value(s)
==6038==    at 0x49A1D1F: PyArray_Scalar (arrayobject.c:1329)
==6038==    by 0x49A3B2E: array_item_nice (arrayobject.c:2001)
==6038==    by 0x49A65E4: array_subscript_nice (arrayobject.c:2931)
==6038==    by 0x80B7394: PyEval_EvalFrame (in /usr/bin/python2.4)
```

This was using numpy '1.0.dev3341', so the affected revision is:

source:/trunk/numpy/core/src/arrayobject.c#3341

I have had a look at the offending line, but it seems ok to me. If you think that these kind of errors are not worth to note, please, tell me and I won't file anymore (unless they would be really harmful).

**Comment by thouis** [comment]
Comment in Trac by atmention:teoliphant, 2006-10-23

These are good things to report.  Could you try it on a recent SVN version of NumPy?

**Comment by thouis** [comment]
Comment in Trac by atmention:FrancescAlted, 2006-11-04

I've finally tried this out. I was getting the same problem with r3428.

I've made some research, and it seems that this only appears when using arrays coming from numarray and converted to !NumPy objects using the array protocol. If I disable the numarray package in !PyTables tests (i.e. conversions through the array protocol doesn't occur), this problem doesn't show up.

I don't know for sure, but maybe the numarray implementation of the array protocol is flawed (using Numeric for doing conversions seems to work well). So, by judging the valgrind message, the most probable thing that is happening is that descr->elsize is not being initialized under some situations (like converting special numarray arrays to numpy).

Sorry for not being able to provide an example of that (the code is too messy, and I'm too lazy to look into it). One possibility is that this problem would be related with:

http://sourceforge.net/tracker/index.php?func=detail&aid=1565683&group_id=1369&atid=450446

because this example was extracted from a similar test.

Anyway, provided that the problem is not in !NumPy itself (unless the problem could be a defect in !NumPy implementation of the array interface), maybe the best should be to close this ticket.

Regards,
